app/src/pages/Lobby/WaitingRoom.tsx

```diff
@@ -63,12 +75,52 @@ function WaitingRoom() {
                   })
                 })
               )
-              await updateGameState({
-                variables: {
-                  id: currentGame.id,
-                  state: GameStateEnum.CardSubmission,
-                },
-              })
+              if (
+                props.cardPlayStyle ===
+                  GameCardPlayStyleEnum.HostProvidesWords &&
+                props.wordList
+              ) {
+                await loadWords({
+                  variables: {
+                    objects: compact(
+                      props.wordList.split(",").map((word) => word.trim())
+                    ).map((word) => {
+                      return {
+                        word,
+                        game_id: currentGame.id,
+                        player_id: currentPlayer.id,
+                        is_allowed: true,
+                      }
+                    }),
+                  },
+                })
+                const players = teamsWithSequence(currentGame.players)
+                await updateAllPlayers({
+                  variables: {
+                    gameId: currentGame.id,
+                    players: players.map(({ id, team, team_sequence }) => ({
+                      id,
+                      team,
+                      team_sequence,
+                    })),
+                  },
+                })
+                updateGameState({
+                  variables: {
+                    id: currentGame.id,
+                    state: GameStateEnum.TeamAssignment,
+                  },
+                })
```

Comment:
bit hacky, but works for now - we can extract out some chained mutations as a custom hook